### PR TITLE
[pdpix] Demikernel in Thread Local Storage

### PR DESCRIPTION
**Pull request by ppenna**
## Description

This PR moves Demikernel to thread local storage.

### CODE REVIEW

File: src/rust/demikernel/bindings.rs [hunk 1]
```diff
@@ -86,25 +88,28 @@ pub extern "C" fn demi_init(argc: c_int, argv: *mut *mut c_char) -> c_int {
...
 
     // Check if demikernel has already been initialized and return
-    match unsafe { DEMIKERNEL.try_borrow() } {
-        Ok(libos) => match *libos {
-            Some(_) => return libc::EEXIST,
-            None => (),
-        },
-        Err(e) => panic!("{:?}", e),
+    let ret: i32 = DEMIKERNEL.with(|demikernel| match *demikernel.borrow() {
+        Some(_) => -libc::EEXIST,
```
Comment (kyleholohan) [inline src/rust/demikernel/bindings.rs:92]: Why are you negating the return code?

Comment (kyleholohan) [reply src/rust/demikernel/bindings.rs:92]: Just double-checked, and libos.h says:
> On failure, a positive error code is returned instead.

**Review by kyleholohan — APPROVED** [review]
Approved, pending comment.

File: src/rust/demikernel/bindings.rs (hunk 1 above)
Comment (ppenna) [reply src/rust/demikernel/bindings.rs]: Thanks.

**Pull request merged**